### PR TITLE
Integrate CircleCI

### PR DESCRIPTION
**Pull request by keradus**
Replaces #2893

Status is failing due to incompatibilities of project to be executed on OSX.

Min requirement to merge this PR:
- [x] bump used PHP from 5.5 to at least 5.6

### CODE REVIEW

**Comment by keradus** [comment]
@aidantwoods, @GrahamCampbell 
as suggested in linked PR, let us try CircleCI for OSX testing.
Do you want to propose improvements to configuration ? (eg faster build, tests on PHP matrix instead of only one version)

**Comment by GrahamCampbell** [comment]
> tests on PHP matrix instead of only one version

Best way to approach that is using circleci v2 I think.

**Comment by GrahamCampbell** [comment]
Instead of all the inbrowser config from v1.

**Comment by keradus** [comment]
> Best way to approach that is using circleci v2 I think.

During setup, they claim v2 is not available for osx yet

> Instead of all the inbrowser config from v1.

how you find config file to be in-browser config?

**Comment by GrahamCampbell** [comment]
> how you find config file to be in-browser config?

In v1, there are settings in the browser for some things. They are totally ignored on v2. The in-repo config is truly only used.

**Comment by keradus** [comment]
again, any suggestion to improve the config having in mind that **v2 is not yet available for osx** ?

File: circle.yml [hunk 1]
```diff
@@ -14,6 +14,7 @@ dependencies:
         - brew tap homebrew/homebrew-php
         - brew install php71
         - curl -sS https://getcomposer.org/installer | php
+        - php composer.phar global show -ND 2>&1 | grep "hirak/prestissimo" || php composer.phar global require --no-interaction --no-progress --optimize-autoloader hirak/prestissimo
```
Comment (aidantwoods) [inline circle.yml]: `composer global info hirak/prestissimo &> /dev/null`?

Comment (keradus) [inline circle.yml]: what for?
if you want to, sent PR to make this change in all CIs

Comment (aidantwoods) [inline circle.yml]: > what for?

Saves invoking `grep` ;p it's fine either way I guess

**Comment by keradus** [comment]
Thank you @aidantwoods.

**Comment by keradus** [comment]
btw,
CircleCI - PHP 7.1, 3m14s
Travis - PHP 7.0, 2m32s

**Comment by aidantwoods** [comment]
(points out that brew is a huge bottleneck)

Circle seems to run `phpunit` pretty damn fast though

screen shot 2017-07-23 at 18 04 15
screen shot 2017-07-23 at 18 05 36

**Comment by keradus** [comment]
Form Travis 7.1:
> Time: 27.64 seconds, Memory: 90.00MB

**Comment by aidantwoods** [comment]
Hm, must've looked at a different build to you (screencaps are Circle on 7.1, travis on 7.0).

**Pull request closed without merging**